addons/mail/static/src/components/thread_needaction_preview/thread_needaction_preview.js

```diff
@@ -0,0 +1,131 @@
+odoo.define('mail/static/src/components/thread_needaction_preview/thread_needaction_preview.js', function (require) {
+'use strict';
+
+const components = {
+    MessageAuthorPrefix: require('mail/static/src/components/message_author_prefix/message_author_prefix.js'),
+    PartnerImStatusIcon: require('mail/static/src/components/partner_im_status_icon/partner_im_status_icon.js'),
+};
+const useStore = require('mail/static/src/component_hooks/use_store/use_store.js');
+const mailUtils = require('mail.utils');
+
+const { Component } = owl;
+const { useRef } = owl.hooks;
+
+class ThreadNeedactionPreview extends Component {
+
+    /**
+     * @override
+     */
+    constructor(...args) {
+        super(...args);
+        useStore(props => {
+            const thread = this.env.models['mail.thread'].get(props.threadLocalId);
+            const mainThreadCache = thread ? thread.mainCache : undefined;
+            let lastNeedactionMessageAuthor;
+            let lastNeedactionMessage;
+            let threadCorrespondent;
+            if (thread) {
+                lastNeedactionMessage = mainThreadCache.lastNeedactionMessage;
+                threadCorrespondent = thread.correspondent;
+            }
+            if (lastNeedactionMessage) {
+                lastNeedactionMessageAuthor = lastNeedactionMessage.author;
+            }
+            return {
+                isDeviceMobile: this.env.messaging.device.isMobile,
+                lastNeedactionMessage: lastNeedactionMessage ? lastNeedactionMessage.__state : undefined,
+                lastNeedactionMessageAuthor: lastNeedactionMessageAuthor
+                    ? lastNeedactionMessageAuthor.__state
+                    : undefined,
+                thread: thread ? thread.__state : undefined,
+                threadCorrespondent: threadCorrespondent
+                    ? threadCorrespondent.__state
+                    : undefined,
+            };
+        });
+        /**
+         * Reference of the "mark as read" button. Useful to disable the
+         * top-level click handler when clicking on this specific button.
+         */
+        this._markAsReadRef = useRef('markAsRead');
+    }
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Get the image route of the thread.
+     *
+     * @returns {string}
+     */
+    image() {
+        if (this.thread.moduleIcon) {
+            return this.thread.moduleIcon;
+        }
+        if (this.thread.correspondent) {
+            return `/web/image/res.partner/${this.thread.correspondent.id}/image_128`;
+        }
+        if (this.thread.model === 'mail.channel') {
+            return `/web/image/mail.channel/${this.thread.id}/image_128`;
+        }
+        return '/mail/static/src/img/smiley/avatar.jpg';
+    }
```

Comment:
I don't like this being duplicated in different components, maybe it's time to "image" compute in thread?

Reply:
Needaction preview is quite unique in that it uses `moduleIcon` most of the time.
It makes sense to move most logic in models, but this one takes (very relatively) too much time to be done in this PR or before merge tomorrow.